src/OrchardCore/OrchardCore.Media.Core/DefaultMediaFileStoreCacheFileProvider.cs

```diff
@@ -56,9 +56,19 @@ public async Task SetCacheAsync(Stream stream, IFileStoreEntry fileStoreEntry, C
                 Directory.CreateDirectory(directory);
             }
 
-            using (var fileStream = File.Create(cachePath))
+            // A file download may fail, so a partially downloaded file should be deleted so the next request can reprocess.
+            try
+            {
+                using (var fileStream = File.Create(cachePath))
+                {
+                    await stream.CopyToAsync(fileStream, StreamCopyBufferSize, cancellationToken);
+                }
+            }
+            catch (Exception ex)
             {
-                await stream.CopyToAsync(fileStream, StreamCopyBufferSize, cancellationToken);
+                _logger.LogError(ex, "Error saving file {Path}", cachePath);
+                File.Delete(cachePath);
```

Comment:
Check if the file exists. And do a try catch also on the File.Delete, just logging the error.
Maybe add a time based token too in case the deletion takes too long (lock, or file system issue).

Reply:
ok. have one. didn't add a token because it felt like it would need a file system watcher, which seems like more than what we want. The issue looks to be less around file system, and more around network failures during transit from blob storage. (so partially completed). 
We might be able to track that better through the `BlobFileStore` but I think this is the better place to handle it

Reply:
You can create a custom cancelation token that is based on time, the idea is that if the file is locked and you can't delete it, it won't block for too long: https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.cancelafter?view=net-5.0#System_Threading_CancellationTokenSource_CancelAfter_System_Int32_